.github/workflows/lint-and-test.yml

```
@@ -52,3 +51,14 @@ jobs:
 
       - name: Run tests
         run: poetry run pytest
+
+      # This is necessary to prevent the following error when installing the package in CI:
+      # "RuntimeError: This is a shallow repository, so Dunamai may not produce the correct version."
```

Could we change this to be a bit more descriptive? dunamai is a transitive dependency, so the origin of this error is a bit opaque. Something like this:
```suggestion
      # Fetching the full git history is necessary because poetry's dynamic versioning extension
      # requires it, and this extension is run even when the package is pip-installed locally (as we do below).
```